PebbleSharp.Core/Install/InstallClient.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Linq;
+using System.Collections.Generic;
+using PebbleSharp.Core.Responses;
+using PebbleSharp.Core.Bundles;
+using PebbleSharp.Core.BlobDB;
+
+namespace PebbleSharp.Core.Install
+{
+	public class InstallClient
+	{
+		private Pebble _pebble;
+		public InstallClient(Pebble pebble)
+		{
+			_pebble = pebble;
+		}
+		public async Task InstallAppAsync( AppBundle bundle, IProgress<ProgressValue> progress = null )
+		{
+
+			string version = _pebble.Firmware.Version;
+			version = version.Replace("v", "");
+			var components = version.Split(new char[] { '.','-' }, StringSplitOptions.RemoveEmptyEntries);
+
+			int i;
+			IList<int> versionComponents = components.Where(x=>int.TryParse(x,out i)).Select(x => int.Parse(x)).ToList();
```

Comment:
Rather than trying to force it down a LINQ expression, I would do this inside of a foreach to avoid the double int conversion.
```C#
IList<int> versionComponents = new List<int>();
foreach(var component in components)
{
    int i;
    if (int.TryParse(component, out i)
    {
        versionComponents.Add(i);
    }
}
```